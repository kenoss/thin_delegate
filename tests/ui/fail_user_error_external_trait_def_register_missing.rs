#[thin_delegate::external_trait_def]
mod __external_trait_def {
    pub trait AsRef<T: ?Sized> {
        /// Converts this type into a shared reference of the (usually inferred) input type.
        #[stable(feature = "rust1", since = "1.0.0")]
        fn as_ref(&self) -> &T;
    }
}

fn main() {}
