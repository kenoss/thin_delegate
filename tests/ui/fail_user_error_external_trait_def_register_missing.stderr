error: #[thin_delegate::register] missing for trait
 --> tests/ui/fail_user_error_external_trait_def_register_missing.rs:3:5
  |
3 | /     pub trait AsRef<T: ?Sized> {
4 | |         /// Converts this type into a shared reference of the (usually inferred) input type.
5 | |         #[stable(feature = "rust1", since = "1.0.0")]
6 | |         fn as_ref(&self) -> &T;
7 | |     }
  | |_____^
